### PR TITLE
enable daemonset

### PR DESCRIPTION
**Pull request by pweil-**
We would like to use daemon sets to deploy the atomic registry.  This PR adds the service account, policy, and starts the controller when the experimental api is enabled. 

@deads2k @soltysh @liggitt

### CODE REVIEW

File: pkg/cmd/server/kubernetes/master.go [hunk 1]
```diff
@@ -190,6 +191,13 @@ func (c *MasterConfig) RunHPAController(oc *osclient.Client, kc *client.Client,
 	podautoscaler.Run(c.ControllerManager.HorizontalPodAutoscalerSyncPeriod)
 }
 
+func (c *MasterConfig) RunDaemonSetsController(client *client.Client) {
+	controller := daemon.NewDaemonSetsController(client, c.ControllerManager.ResyncPeriod)
+	// TODO this first arg should come from ConcurrentDSCSyncs configuration
```
Comment (liggitt) [inline pkg/cmd/server/kubernetes/master.go]: that var is coming in the rebase

Comment (pweil-) [inline pkg/cmd/server/kubernetes/master.go]: :+1:

**Comment by pweil-** [comment]
[test][extended:core]

File: test/extended/core.sh [hunk 2]
```diff
@@ -51,7 +51,6 @@ SKIP_TESTS=(
   DNS                     # Can't depend on kube-dns
   kube-ui                 # Not installed by default
   DaemonRestart           # Experimental mode not enabled yet
```
Comment (soltysh) [inline test/extended/core.sh]: Isn't that test also related to DaemonSet?

Comment (kargakis) [inline test/extended/core.sh]: hm, no it's not. The comment next to it is wrong.

Comment (soltysh) [inline test/extended/core.sh]: I've looked into the code and it looks like it's testing restarting nodes and if the cluster will react approprietly, so I guess this should be tested.

Comment (kargakis) [inline test/extended/core.sh]: If we don't mind https://github.com/kubernetes/kubernetes/issues/19023 and https://github.com/kubernetes/kubernetes/issues/17829 :)

**Comment by deads2k** [comment]
lgtm

File: test/extended/core.sh (hunk 2 above)
Comment (soltysh) [inline test/extended/core.sh]: ;)

**Comment by pweil-** [comment]
womp womp...
```
There was an error talking to AWS. The error message is shown
below:

Error: EC2 Machine is not available
```

re[test]

**Comment by soltysh** [comment]
> There was an error talking to AWS.

This is why you should stay away from stuff like that on Friday :wink:

**Comment by pweil-** [comment]
Failed parsing docker version due to strings in the version https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin/9187/console

Created PR: https://github.com/openshift/origin/pull/6792

**Comment by pweil-** [comment]
re[test]

**Comment by pweil-** [comment]
[testonlyextended][extended:core("Daemon set")]

**Comment by pweil-** [comment]
re[test]

**Comment by deads2k** [comment]
Is this waiting on anything else?

File: pkg/cmd/server/kubernetes/master.go (hunk 1 above)
Comment (liggitt) [inline pkg/cmd/server/kubernetes/master.go]: rebase is in the merge queue, pick up the config var before merging this

**Comment by pweil-** [comment]
> Is this waiting on anything else?

I want to get a run of the DaemonSets extended tests here if I can (ran locally fine).   It skipped the test on the last run due to casing I'm assuming so I've changed it to " sets" to try and pick it up on this run.  Other than that, just the rebase var and good to go

**Comment by pweil-** [comment]
[testonlyextended][extended:core(aemon set)]

**Comment by pweil-** [comment]
[testonlyextended][extended:core]

**Pull request closed without merging**